src/linux/mgsocket.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2016-2020 Memgraph Ltd. [https://memgraph.com]
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef MGCLIENT_LINUX_MGSOCKET_H
+#define MGCLIENT_LINUX_MGSOCKET_H
+
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <netinet/ip.h>
+#include <netinet/tcp.h>
+#include <poll.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+// Include top level mgsocket file because the implementation file needs it.
+#include <mgsocket.h>
```

Comment:
oh, this is awkward now... I think you can include just the top level mgsocket header file in all the implementations, platform specific stuff is resolved in it. The problem is the same name as the platform specific and top level header file. :thinking: 
Or, we can leave it as is.

Reply:
Also, this is and should it be only used in linux/mgsocket.c so why not include the specific headers directly and remove platform specific mgsocket headers?
Same for other platforms.

Reply:
Yep, I'll just revert to how it was and include everything inside `src/mgsocket.h`. That's I think better.